### PR TITLE
MAGE-1050: category search test included.

### PR DESCRIPTION
**Pull request by mrahman3177**
**Summary**

Category search integration test included.
Screenshot 2024-10-09 at 7 55 36 PM

### CODE REVIEW

File: Test/Integration/Search/SearchTest.php [hunk 1]
```diff
@@ -8,18 +8,34 @@
...
+        list($results, $totalHits, $facetsFromAlgolia) = $this->helper->getSearchResult('', 1);
+        $this->assertNotEmpty($results);
+    }
 
+    public function testCategorySearch()
+    {
+        list($results, $totalHits, $facetsFromAlgolia) = $this->helper->getSearchResult('', 1, [
+            'facetFilters' => ['categoryIds:' . $this->assertValues->expectedCategory]
+        ]);
         $this->assertNotEmpty($results);
```
Comment (cammonro) [inline Test/Integration/Search/SearchTest.php:39]: Can we verify that the number of products retrieved with the category facet matches was is expected - perhaps against a collection through `CategoryRepository` and then verify parity with the Algolia index? i.e. counts should be identical. Keep in mind other settings such as whether included in menu etc'.

File: Test/Integration/Search/SearchTest.php [hunk 2]
```diff
@@ -8,18 +8,34 @@
...
+        $this->productIndexer->executeFull();
         $this->algoliaHelper->waitLastTask();
+    }
 
-        /** @var Data $helper */
-        $helper = $this->getObjectManager()->create(Data::class);
-        list($results, $totalHits, $facetsFromAlgolia) = $helper->getSearchResult('', 1);
+    public function testSearch()
+    {
+        list($results, $totalHits, $facetsFromAlgolia) = $this->helper->getSearchResult('', 1);
```
Comment (cammonro) [inline Test/Integration/Search/SearchTest.php:30]: Let's broaden our test cases a little to actually verify that expected products, skus, and colors are found against certain assertion values and not just that there are results.

**Review by cammonro — CHANGES_REQUESTED** [review]
Please see comments in PR and Jira.

File: Test/Integration/Search/SearchTest.php [hunk 3]
```diff
@@ -5,21 +5,101 @@
...
+        $this->assertNotEmpty($product->getSku(), "SKU search result item couldn't find in the DB");
+        // Query word exists title
+        $this->assertEquals($sku, $product->getSku(), "Query SKU doesn't match with product SKU");
+    }
+
+    public function testCategorySearch()
+    {
+        // Get products by categoryId
+        list($results, $totalHits, $facetsFromAlgolia) = $this->search('', 1, [
+            'facetFilters' => ['categoryIds:' . $this->assertValues->expectedCategory]
```
Comment (damcou) [inline Test/Integration/Search/SearchTest.php:65]: `expectedCategory` is in fact the expected number of categories indexed in the Algolia dashboard and not an actual ID . I guess `$this->assertValues->expectedCategory` is quite missleading 😄 . 

Feel free to use a `const` in the class to arbitrary search for a category id (like we did in newly created tests) : https://github.com/algolia/algoliasearch-magento-2/blob/release/3.14.3/Test/Integration/Category/MultiStoreCategoriesTest.php#L31

File: Test/Integration/Search/SearchTest.php [hunk 4]
```diff
@@ -5,21 +5,101 @@
...
+        ]);
+        // Category filter returns result
+        $this->assertNotEmpty($results, "Category filter didn't return result");
+
+        $collection = $this->objectManager->create(\Magento\Catalog\Model\ResourceModel\Product\Collection::class);
+        $collection
+            ->addAttributeToSelect('*')
+            ->addAttributeToFilter('status',\Magento\Catalog\Model\Product\Attribute\Source\Status::STATUS_ENABLED)
+            ->addAttributeToFilter('visibility', \Magento\Catalog\Model\Product\Visibility::VISIBILITY_BOTH)
+            ->addCategoriesFilter(["in" => $this->assertValues->expectedCategory])
```
Comment (damcou) [inline Test/Integration/Search/SearchTest.php:75]: To be replaced here as well :)

**Review by damcou — COMMENTED** [review]
Tested locally and it works as expected ! 🙌 
Just need to make 2 small updates I think, good job overall !

**Review by damcou — APPROVED** [review]
LGTM, good work ! 🙌

**Pull request merged**